design/gateway-apis-implementation.md

```diff
@@ -163,23 +169,26 @@ The output of this watcher is Envoy configuration.
 ## Detailed Design
 
 ### Configuration
-Contour will have an entry added to the config file for the Gateway it should watch.
+Contour will have an entry added to the config file for the `ControllerName` it should watch.
+The `ControllerName` takes the format of `projectcontour.io/<namespace>/contour` which requires the `<namespace>` param to be specified matching the namespace where Contour is deployed. 
 
 This will look like this:
 ```yaml
 gateway:
-  name: gatewayname
-  namespace: gatewaynamespace
+  controllerName: 
+  name: gatewayname  # <--- deprecate
+  namespace: gatewaynamespace  # <-- deprecate
 ```
 
-The current flag `--experimental-service-apis` will be removed.
-The boolean it represents will be deduced from the following rules:
-- the gateway is configured in the config file.
-- the referenced gateway exists.
+Contour will look for a Gateway + GatewayClass that matches the `ControllerName` field specified in the configuration file.
+If it does not find a matching Gateway/GatewayClass, then it will begin processing GatewayAPI resources once they are available. 
+
+Should two different Gateway or GatewayClasses be found matching the same `ControllerName`, then Contour will utilize the oldest created object and set a status on the other(s) which describes that object as being rejected.
```

Comment:
This should specify what Condition will be emitted, what Reason should be set, and so on.